lib/clients/GcpClient.py

```diff
@@ -0,0 +1,555 @@
+                                 .format(log_prefix, blob_to_upload_path, blob_target_name, self.CONTAINER))
+                return True
+            except Exception as error:
+                message = '{} ERROR: blob_to_upload={}, blob_target_name={}, container={}\n{}'.format(
+                    log_prefix, blob_to_upload_path, blob_target_name, self.CONTAINER, error)
+                self.logger.error(message)
+                raise Exception(message)
+        else:
+            message = '{} ERROR: blob_to_upload={}, blob_target_name={}, container={}\n{}'.format(
+                log_prefix, blob_to_upload_path, blob_target_name, self.CONTAINER, "Container not found or accessible")
+            self.logger.error(message)
+            raise Exception(message)
+
+    def _download_from_blobstore(self, blob_to_download_name, blob_download_target_path, chunk_size=None):
+        """Download file from blobstore.
+        :type chunk_size: int
+        :param chunk_size: If file size if greater than 5MB, it is recommended that, 
+                           chunked downloads should be used.
+                           To do so, pass chunk_size param to this function.
+                           This must be a multiple of 256 KB per the API specification.
+        """
+        log_prefix = '[Google Cloud Storage] [DOWNLOAD]'
+
+        if self.container:
+            self.logger.info('{} Started to download the tarball to target.'.format(
+                log_prefix, blob_download_target_path))
+            try:
+                blob = Blob(blob_to_download_name,
+                            self.container, chunk_size=chunk_size)
+                blob.download_to_filename(blob_download_target_path)
+                self.logger.info('{} SUCCESS: blob_to_download={}, blob_target_name={}, container={}'
+                                 .format(log_prefix, blob_to_download_name, self.CONTAINER,
+                                         blob_download_target_path))
+                return True
+            except Exception as error:
+                message = '{} ERROR: blob_to_download={}, blob_target_name={}, container={}\n{}'.format(
+                    log_prefix, blob_to_download_name, blob_download_target_path, self.CONTAINER, error)
+                self.logger.error(message)
+                raise Exception(message)
+        else:
+            message = '{} ERROR: blob_to_download={}, blob_target_name={}, container={}\n{}'.format(
+                log_prefix, blob_to_download_name, blob_download_target_path, self.CONTAINER, "Container not found or accessible")
+            self.logger.error(message)
+            raise Exception(message)
+
+    '''
+    def _download_from_blobstore_and_pipe_to_process(self, process, blob_to_download_name, segment_size):
+        # TODO Currently not used for GCP. Implement if needed.
+        pass
+    '''
+
+    def wait_for_operation(self, operation_id, zonal_operation):
```

Comment:
Better name ? `get_operation_status`
No actual waiting is done in this function.

Reply:
Yes, will change it to 'get_operation_status'.